python/cuml/experimental/fil/fil.pyx

```diff
@@ -1228,6 +1263,106 @@ class ForestInference(UniversalBase, CMajorInputTagMixin):
             any power of 2, but little benefit is expected above a chunk size
             of 512.
         """
+        chunk_size = self._get_chunk_size(chunk_size)
         return self.forest.predict(
             X, predict_type="per_tree", preds=preds, chunk_size=chunk_size
         )
+
+    def optimize(
```

Comment:
Instead of making this a method that modifies the estimator in place, I would make this a functor that operates on an estimator. I'm thinking of the pattern we typically use for hyper-parameter optimization and pipelining.

So, the pattern would look more like this:
```python
estimator = ForestInference.load(...)
optimized_estimator = ForestInferenceTuner(sample_data).optimize(estimator)
optimized_estimator.predict(data)
```
That would allow the short-pattern of:
```python
estimator = ForestInferenceOptimizer(sample_data).optimize(ForestInference.load(...))
```
I think that would make it easier to make this part of a pipeline and might avoid some headaches relating to modifying the estimator in place.

Reply:
Following offline discussion, we're going to keep the `.optimize` method but look at a more general-purpose performance optimization class for use with all estimators.